### PR TITLE
[9.x] New methods of eloquent - createAnd and updateAnd

### PR DESCRIPTION
**Pull request by m-mohammadi1**
Most of the time we need to do this
```php
$model = User::create[...]);

if ($model) {
    // do something ...
}
```
for this one we could just use this shortcut
```php
$model = User::createAnd([...], function ($created_model) {
      // do something ...
  });
```

And for the same reason we could use updateAnd on Model and Eloquent Builder.
For Model
```php
$model = Model::find(1);
$model->updateAnd([...], function ($updated_model) {
    // more code ...
});
```
on Builder we can use this
```php
$models = Model::query();

$models->updateAnd([...], function ($updated_models) {
    // code more ...
});
```

I think it is good practice we do every thing related to an stuff in a area that relates to it.
and these are like an inline model observers that we can use in our logic.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -608,6 +608,44 @@ public function firstOr($columns = ['*'], Closure $callback = null)
         return $callback();
     }
 
+    /**
+     * Create record in database and call user given callback with created model.
+     *
+     * @param  array  $attributes
+     * @param  Closure  $callback
+     * @return mixed|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php:616]: `mixed` cannot exist next to any other type in a union. `null` is already contained in `mixed`.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by rodrigopedra** [comment]
Why not using `tap()`?

~~~php
\tap(new Model, function ($model) {
 // do something ...
});
~~~

reference: https://laravel.com/docs/9.x/helpers#method-tap

File: src/Illuminate/Database/Eloquent/Builder.php (hunk 1 above)
Comment (m-mohammadi1) [reply src/Illuminate/Database/Eloquent/Builder.php:616]: > Why not just use `tap()`?
> 
> ```
> \tap(new Model, function ($model) {
>  // do something ...
> });
> ```
> 
> reference: https://laravel.com/docs/9.x/helpers#method-tap
I think this is more user friendly and can help new users and actually it is obvious what it does.

Comment (m-mohammadi1) [reply src/Illuminate/Database/Eloquent/Builder.php:616]: > `mixed` cannot exist next to any other type in a union. `null` is already contained in `mixed`.

yes I should fix it.

**Comment by m-mohammadi1** [comment]
> Why not just use `tap()`?
> 
> ```
> \tap(new Model, function ($model) {
>  // do something ...
> });
> ```
> 
> reference: https://laravel.com/docs/9.x/helpers#method-tap

I think this is more user friendly and can help new users and actually it is obvious what it does.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**